core/src/snapshot_packager_service.rs

```diff
@@ -87,7 +87,14 @@ impl SnapshotPackagerService {
 
             // `storage_path` - The file path where the AppendVec itself is located
             // `output_path` - The directory where the AppendVec will be placed in the staging directory.
-            symlink::symlink_dir(storage_path, output_path)?;
+            let storage_path =
+                fs::canonicalize(storage_path).expect("Could not get absolute path for accounts");
```

Comment:
Is there a rule of thumb for panicking vs returning an error? I opted for a panic since it seemed like this should never fail

Reply:
Ideally:
1. If the error is due to user error or other untrusted input, return an error so we can cleanly exit or otherwise recover. 
2. If the error is due to an internal bug or an assumption we make in the code that is suddenly not true, `panic!`.

Unfortunately we still have many cases of (1) that manifest as panics in the code

Reply:
Can we `fs::canonicalize()` once during snapshot initialization so we have a more clean path to error back to the user?

Reply:
@mvines that's pretty much this: https://github.com/solana-labs/solana/pull/7294/files#diff-1f422d7e73618aca7bce9935356de012R575 right?

Reply:
Oh actually I need to check the `--accounts` arg too

Reply:
@mvines cool if I fix `--accounts` in a follow up? Looks like it will require a refactor

Reply:
Sure